ts/src/base/Exchange.ts

```diff
@@ -2837,6 +2837,10 @@ export default class Exchange {
         return cleanStructure;
     }
 
+    isSpotMargin (marketType: Str = undefined, marginMode: Str = undefined): boolean {
+        return (marketType === 'margin') || (((marketType === undefined) || (marketType === 'spot')) && (marginMode !== undefined));
```

Comment:
why checking marketType == undefined here? if marketType is undefined then it's hard to take conclusions about this (if this marketType arrives from handleMarketTypeAndParams... then marketType should not be undefined)

Reply:
but what should happen in case if marketType is undefined and marginMode defined?

Reply:
how could that happen? specially if you're inside createOrder or something you will have a symbol.

Reply:
> how could that happen? specially if you're inside createOrder or something you will have a symbol.

eg. `createOrder ('XYZUSDT', ...... {marginMode:'cross'})` (where `XYZUSDT` is exchagne specifci symbol, which is added recently but not loaded in `loadMarkets`.

or

`fetchOpenOrders (undefined, {marginMode:'cross'})`

I just try to think of those peripherical cases, however, if you still think we shouldnt have it, i'm going to remove it and push atm

Reply:
so my question - do we expect that there would be cases when marketType can be undefiend, but `marginMode` defined? if not, then we can even add:
```
if (marginMode !== undefined && marketType === undefined) {
    throw ArgumentsRequired (" ...whatever ");
}
```
but if we expect such cases, then what should we do?

Reply:
@ttodua the marketType arriving to isSpotMargin should come from `handleMarketTypeAndParams` so it should never be undefined

Reply:
> so it should never be undefined

as I've mentioned above, there could be cases (like incoming new market which was not loaded during `loadMarkets`) however those are rarest cases. 
ok, so the PR is as you say, lmk if i miss smth